tests/Integration/Behaviour/Features/Scenario/Order/order_gift_cart_rule.feature

```diff
@@ -1169,3 +1169,156 @@ Feature: Order from Back Office (BO)
       | total_paid_tax_incl      | 0.00  |
       | total_paid               | 0.00  |
       | total_paid_real          | 0.00  |
+
+  Scenario: I have a cart rule that adds a free gift when delivery address is in the USA,
+            when I change this delivery address, the free gift is removed
+
+    Given country "US" is enabled
+    And country "FR" is enabled
+    And there is a product in the catalog named "Product 12345" with a price of 12.0 and 100 items in stock
+    And there is a product in the catalog named "Gift product" with a price of 13.0 and 100 items in stock
+    # Create a cart rule : No cade + no conditions + free gift = demo_6
+    And there is a cart rule named "cartRuleFreeGift" that applies no discount with priority 1, quantity of 100 and quantity per user 100
+    And cart rule "cartRuleFreeGift" has no discount code
+    And cart rule "cartRuleFreeGift" offers a gift product "Gift product"
+    And cart rule "cartRuleFreeGift" is restricted to country "US"
+    # Make an order
+    And I create an empty cart "dummy_cart" for customer "testCustomer"
+    And I select "US" address as delivery and invoice address for customer "testCustomer" in cart "dummy_cart"
+    And I add 1 products "Product 12345" to the cart "dummy_cart"
+    And I add order "bo_order1" with the following details:
+      | cart                | dummy_cart                 |
+      | message             | test                       |
+      | payment module name | dummy_payment              |
+      | status              | Awaiting bank wire payment |
+    Then order "bo_order1" should have 2 products in total
+    And order "bo_order1" should have 0 invoice
+    And order "bo_order1" should have 1 cart rule
+    And order "bo_order1" should have following details:
```

Comment:
More often than not, sadly 😅 , the order totals are correct but not the OrderDetails So since this use case is quite complex it might be interesting to add two steps to validate each Order product's details